src/util/psbt/mod.rs

```diff
@@ -16,6 +21,138 @@ pub mod serialize;
 mod psbtmap;
 pub use self::psbtmap::{PsbtMap, PsbtGlobal, PsbtInput, PsbtOutput};
 
+/// A Partially Signed Transaction.
+#[derive(Debug, Clone)]
+pub struct PartiallySignedTransaction {
+    /// The key-value pairs for all global data.
+    global: PsbtGlobal,
+    /// The corresponding key-value map for each input in the unsigned
+    /// transaction.
+    inputs: Vec<PsbtInput>,
+    /// The corresponding key-value map for each output in the unsigned
+    /// transaction.
+    outputs: Vec<PsbtOutput>,
+}
+
+impl PartiallySignedTransaction {
+    /// Create a PartiallySignedTransaction from an unsigned transaction, error
+    /// if not unsigned
+    pub fn from_unsigned_tx(tx: Transaction) -> Result<Self, util::Error> {
+        Ok(PartiallySignedTransaction {
+            inputs: Vec::with_capacity(tx.input.len()),
+            outputs: Vec::with_capacity(tx.output.len()),
```

Comment:
I think this is wrong - search " The transaction format is specified as follows:" in [BIP 174](https://github.com/bitcoin/bips/blob/master/bip-0174.mediawiki#Specification) which says that there must be a map for every input and output. Using `with_capacity()` will create empty vectors, so that serializing a fresh `PartiallySignedTransaction` will result in an invalid PSBT.

I think providing empty maps with `vec![Default::default(); tx.input.len()]` should be fine; then an Updater can add actual info.